src/CBLCollection_Internal.hh

```diff
@@ -127,7 +131,82 @@ public:
                                         (C4QueryLanguage)config.expressionLanguage,
                                         kC4FullTextIndex, &options);
     }
+    
+#ifdef COUCHBASE_ENTERPRISE
+    
+    void createVectorIndex(slice name, CBLVectorIndexConfiguration config) {
+        if (!config.expression.buf) {
+            C4Error::raise(LiteCoreDomain, kC4ErrorInvalidParameter, "expression is required.");
+        }
+            
+        if (config.dimensions < 2 || config.dimensions > 2048) {
+            C4Error::raise(LiteCoreDomain, kC4ErrorInvalidParameter, "dimensions must be >= 2 and <= 2048.");
+        }
+        
+        if (config.centroids < 1 || config.centroids > 64000) {
+            C4Error::raise(LiteCoreDomain, kC4ErrorInvalidParameter, "centroids must be >= 1 and <= 64000.");
+        }
+        
+        // Use default minTrainingSize if minTrainingSize is not specified:
+        if (config.minTrainingSize == 0) {
+            config.minTrainingSize = 25 * config.centroids;
+        }
+        
+        // Use default maxTrainingSize if maxTrainingSize is not specified:
+        if (config.maxTrainingSize == 0) {
+            config.maxTrainingSize = 250 * config.centroids;
+        }
+        
+        if (config.minTrainingSize < 1) {
+            C4Error::raise(LiteCoreDomain, kC4ErrorInvalidParameter, "minTrainingSize must be > 1.");
+        } else if (config.maxTrainingSize < 1) {
+            C4Error::raise(LiteCoreDomain, kC4ErrorInvalidParameter, "maxTrainingSize must be > 1.");
+        } else if (config.minTrainingSize > config.maxTrainingSize) {
+            C4Error::raise(LiteCoreDomain, kC4ErrorInvalidParameter, "minTrainingSize must be <= maxTrainingSize.");
+        }
```

Comment:
I like this more, will have to change it in iOS. Maybe is worth adding how we handle the platform checks in the spec to refer back to it for Beta 2 so we have it consistent.

Reply:
I would like that to be a bit flexible for each platform developer to decide. We can sync about the error messaging though. We had a common project for sharing error message. You can check with Blake.